src/radical/pilot/pytask.py

```diff
@@ -0,0 +1,92 @@
+
+__copyright__ = 'Copyright 2022, The RADICAL-Cybertools Team'
+__license__   = 'MIT'
+
+import inspect
+import functools
+
+from types import Callable
+
+from .utils import serialize_obj, serialize_bson
+from .utils import deserialize_obj, deserialize_bson
+
+
+# ------------------------------------------------------------------------------
+#
+class PythonTask(object):
+
+    # --------------------------------------------------------------------------
+    #
+    def __new__(cls, func, *args, **kwargs):
+        '''
+        We handle wrapped functions here with no args or kwargs.
+
+        Example:
+            import PythonTask
+            wrapped_func   = partial(func_A, func_AB)
+            cud.EXECUTABLE = PythonTask(wrapped_func)
+        '''
+
+        if not inspect.isfunction(func):
+            raise ValueError('task function not callable')
+
+        task = {'func'  : serialize_obj(func),
+                'args'  : args,
+                'kwargs': kwargs}
+        try:
+            return serialize_bson(task)
+
+        except Exception as e:
+            raise ValueError from e
```

Comment:
I still don't understand why the exception is caught and then thrown again.  What is the benefit?  We don't even now if the problem really was an `ValueError` - maybe we ran out of Memory?  I would be in favor of just letting it fall through.